src/data/mv_polynomial/basic.lean

```diff
@@ -292,6 +295,14 @@ finsupp.induction p (suffices P (monomial 0 0), by rwa monomial_zero at this,
                      show P (monomial 0 0), from h1 0 0)
                     (λ a b f ha hb hPf, h2 _ _ (h1 _ _) hPf)
 
+lemma induction_on'' {M : mv_polynomial σ R → Prop} (p : mv_polynomial σ R) (h_C : ∀a, M (C a))
+ (h_add_weak : ∀ (a : σ →₀ ℕ) (b : R) (f : (σ →₀ ℕ) →₀ R),
+ a ∉ f.support → b ≠ 0 → M f → M (single a b + f)) (h_X : ∀p n, M p → M (p * X n)) : M p :=
+have ∀s a, M (monomial s a) := induction_on_monomial h_C h_X,
+finsupp.induction p
+  (by have : M (C 0) := h_C 0; rwa [C_0] at this)
+  h_add_weak
```

Comment:
Multiple issues here:
* bad indents
* the `have` is unused
* `single` is not part of the `mv_polynomial` API, it should be `monomial`:
```suggestion
lemma induction_on'' {M : mv_polynomial σ R → Prop} (p : mv_polynomial σ R) (h_C : ∀ a, M (C a))
  (h_add_weak : ∀ (a : σ →₀ ℕ) (b : R) (f : (σ →₀ ℕ) →₀ R),
    a ∉ f.support → b ≠ 0 → M f → M (monomial a b + f)) (h_X : ∀p n, M p → M (p * X n)) : M p :=
finsupp.induction p (C_0.rec $ h_C 0) h_add_weak
```